src/shared_utilities/SpaceVaryingInputs.jl

```diff
@@ -0,0 +1,133 @@
+  
+Returns the parameter field to be used in the model; appropriate when
+a parameter is defined using a function of the coordinates of the space.
+
+Pass the ``data" as a function `data_function` which takes coordinates as arguments, 
+and  the ClimaCore space of the model simulation.
+"""
+function SpaceVaryingInput(
+    data_function::Function,
+    space::ClimaCore.Spaces.AbstractSpace,
+)
+    model_value = ClimaCore.Fields.zeros(space)
+    coords = ClimaCore.Fields.coordinate_field(space)
+    @. model_value = data_function(coords)
+    return model_value
+end
+
+# 1-D Case
+"""
+    function SpaceVaryingInput(
+        data_z::AbstractArray,
+        data_value::AbstractArray,
+        space::S,
+    ) where {S <: ClimaCore.Spaces.CenterFiniteDifferenceSpace}
+
+Interpolates the variable with values given in `data_value` as a function of
+depths `data_z` to the model grid defined implicitly by `space`. Returns a
+ClimaCore.Fields.Field.
+"""
+function SpaceVaryingInput(
+    data_z::AbstractArray,
+    data_value::AbstractArray,
+    space::S,
+) where {S <: ClimaCore.Spaces.CenterFiniteDifferenceSpace}
+    model_value = ClimaCore.Fields.zeros(space)
+    zvalues = ClimaCore.Fields.coordinate_field(space).z
+    N = length(data_z)
+    model_value .= map(zvalues) do z
+        linear_interpolation(data_z, data_value, z)
+    end
+    return model_value
+end
+
+
+"""
+    SpaceVaryingInputs.SpaceVaryingInput(
+        data_z::AbstractArray,
+        data_values::NamedTuple,
+        space::S,
+        dest_type::Type{DT},
+    ) where {
+        S <: ClimaCore.Spaces.CenterFiniteDifferenceSpace,
+        DT,
+    }
+
+Returns a field of parameter structs to be used in the model; 
+appropriate when the parameter struc values vary in depth.
+
+Note that your struct *must* have a constructor which accepts the 
+values of the named tuple by keyword argument, without any other arguments.
+"""
+function SpaceVaryingInput(
+    data_z::AbstractArray,
+    data_values::NamedTuple,
+    space::S,
+    dest_type::Type{DT},
+) where {S <: ClimaCore.Spaces.CenterFiniteDifferenceSpace, DT}
+    zvalues = ClimaCore.Fields.coordinate_field(space).z
+    data_keys = keys(data_values)
+    model_value = map(zvalues) do z
+        args = map(data_keys) do key
+            key => linear_interpolation(data_z, getproperty(data_values, key), z)
+        end
+        DT(; args...)
+    end
+    return model_value
+end
+
+
+# 2D case (struct version not supported yet)
+"""
+    SpaceVaryingInput(data::PDS, varname::N; space::S) 
+    where {PDS <: FileReader.PrescribedDataStatic, N <: Symbol, S <: ClimaCore.Spaces.SpectralElement2D}
+  
+Returns the parameter field to be used in the model; appropriate when
+a parameter is defined on the surface of the Earth.
+
+Pass the data as a `FileReader.PrescribedDataStatic` object
+as well as the variable name of the variable in the data file, and the ClimaCore space
+of the model simulation.
+"""
```

Comment:
could add a note that this returns a Field of scalars and is analogous to the non-struct 1D case